File: Face_Recognize/Python/tools/camera.py
#!/usr/bin/env python
# -*- coding: UTF-8 -*-

import cv2

def camera():
	capture = cv2.VideoCapture(0)
	index = 1
	while capture.isOpened():
		# 返回一个元组，value和image，value用来检查是否读入成功
		ok, frame = capture.read()
		if not ok:
			break;
		else:
			cv2.imshow("MyPicture", frame)
			gray = cv2.cvtColor(frame, cv2.COLOR_BGR2GRAY)
			# cv2.imshow("MyPicture", gray)
			dst = cv2.resize(gray, (92, 112));
			c = cv2.waitKey(100)
			if c & 0xFF == ord('q'):				# q退出
				break;
			elif c & 0xFF == ord('a'):				# a拍照
				filename = '%d.pgm' % index
				index += 1
				cv2.imwrite(filename, dst)
			else:
				continue
	capture.release()
	cv2.destroyAllWindows()

if __name__ == '__main__':
	camera()

File: Face_Recognize/C++/tools/camera.cpp
#include <iostream>
#include <opencv2/opencv.hpp>
using namespace cv;
using namespace std;

int main(int argc, char const *argv[])
{
	// 创建VideoCapture对象，打开默认摄像机
	VideoCapture capture(0);
	// 检查是否打开摄像头
	if (!capture.isOpened()) {
		cerr << "VideoCapture construct error!" << endl;
		exit(1);
	}

	Mat dst;
	Mat frame;
	Mat changeframe;
	int i = 1;
	while (1) {
		// 从摄像机中获取一个新的框架
		capture >> frame;
		if (frame.empty()) {
			cerr << "New frame is empty" << endl;
			exit(1);
		}

		// 显示灰度图
		imshow("MyPicture", frame);

		// 转为灰度图
		cvtColor(frame, changeframe, CV_BGR2GRAY);

		// 改变图片大小
		resize(changeframe, dst, Size(92, 112), 0, 0, INTER_LINEAR);
		char c = waitKey(100);
		if (c == 97) {										// 'a'开始拍照
			++i;
			string filename = format("%d.jpg", i);			// 定义图片存储的位置
			imwrite(filename, dst);							// 保存灰度图
		}
		else if (c == 27) {									// Esc推出程序
			break;
		}
	}

	return 0;
}
